content/chainguard/chainguard-images/getting-started/php.md

```diff
@@ -223,6 +223,111 @@ In such cases, the last `FROM` section from the Dockerfile is the one that compo
 
 It's worth highlighting that nothing is carried from one stage to the other unless you copy it. That facilitates creating a slim final image with only what's necessary to execute the application.
 
+## 3. Working with the PHP-FPM Image Variant
+
+The `latest-fpm` image variant is suitable for running PHP applications over FastCGI, to be server via a web server such as Nginx. In this section, we'll run a Docker Compose setup using the `latest-fpm` image variant and the Chainguard Nginx image.
+
+The `namegen-api` demo is a variation of the previous demo, but it serves the random name generation over HTTP. The application responds with a JSON payload containing the animal and adjective combination, and accepts an optional `animal` parameter to specify the animal for the final suggested name.
+
+Start by accessing the `namegen-api` demo folder:
+
+```shell
+cd edu-images-demos/php/namegen-api
+```
+
+The `index.php` file contains the following code:
+
+```php
+<?php
+
+$animals = [ 'turtle', 'seagull', 'octopus', 'shark', 'whale', 'dolphin', 'walrus', 'penguin', 'seahorse'];
+$adjectives = [ 'ludicrous', 'mischievous', 'graceful', 'fortuitous', 'charming', 'ravishing', 'gregarious'];
+
+$chosenAdjective = $adjectives[array_rand($adjectives)];
+$chosenAnimal = $_GET['animal'] ?? $animals[array_rand($animals)];
+
+echo json_encode(['animal' => $chosenAnimal, 'adjective' => $chosenAdjective]);
+```
+
+To serve this application over HTTP, we'll use the `latest-fpm` image variant in a Docker Compose setup. The following `docker-compose.yml` is included within the `namegen-api` folder:
+
+```yaml
+services:
+  app:
+    image: cgr.dev/chainguard/php:latest-fpm
+    restart: unless-stopped
+    working_dir: /app
+    volumes:
+      - ./:/app
+
+  nginx:
+    image: cgr.dev/chainguard/nginx
+    restart: unless-stopped
+    ports:
+      - 8000:80
+    volumes:
+      - ./:/app
+      - ./nginx.conf:/etc/nginx/nginx.conf
+```
+
+This Docker Compose setup defines two services: `app` and `nginx`. The `app` service uses the `latest-fpm` image variant and mounts the current directory to the `/app` directory in the container. The `nginx` service uses the Chainguard Nginx image and also mounts the current directory to the `/app` directory in the container, setting it as the document root with a custom configuration. A second volume replaces the default Nginx configuration with our custom one, included as `nginx.conf` in the same directory:
```

Comment:
this paragraph is great! super clear explanation and a smooth transition to the next file

Reply:
thank you! 🙌🏼